src/public/user.ts

```diff
@@ -22,6 +23,8 @@ export const userPublicHandlers: PublicNatsHandler[] = [
 
                     const user = (data.body as unknown) as JSONUser;
 
+                    userSchema.validate(user);
```

Comment:
we should move the validation logic to the private handler.
If it's in the public handler then another micro service calling the private handler will not benefit from the validation logic. 
Also, as a rule of thumb, there should be little to no logic in public handlers, they should delegate all of it to the private handlers. Public handlers are just "facades"